### PR TITLE
Refactor .read() to use requests

### LINKED ISSUE #2852: Replace all urllib / urllib2 imports and uses with `requests`
`urllib2` is old, and a bit clunky -- I've found recently it can't do some of things I needed it to do, and it also causes problem for moving to Python3, and generally seems to make the code more complex.

Instead of re-using it, and adding new features that use the older mechanisms, can we consider refactoring to use `requests` instead?

My PR where I stripped out `urllib2` and `StringIO` etc to enable `HEAD` requests:  #2838

a similar PR that went with using `requests` to avoid old bugs: #2779 

By using `requests` (rather than adding imports from `six`) we should be getting a better tool for making requests, and reducing code complexity, and the number of imports. 

Relates to #2308 since a common use of `simplejson` is to parse `urllib2` responses. `requests` has a builtin `json()` method so an extra import is not required.

### PR DESCRIPTION
**Pull request by aasifkhan7**
Closes #2852
related to #3694 

@cclauss clubbed `.read()` to `.content` or `.text` here

### CODE REVIEW

File: openlibrary/admin/numbers.py [hunk 1]
```diff
@@ -132,10 +131,9 @@ def admin_range__visitors(**kargs):
         sqlitefile = tempfile.mktemp(prefix="sqlite-")
         url = "http://www.archive.org/download/stats/numUniqueIPsOL.sqlite"
         logging.debug("  Downloading '%s'", url)
-        sqlite_contents = urllib.request.urlopen(url).read()
-        f = open(sqlitefile, "w")
-        f.write(sqlite_contents)
-        f.close()
+        sqlite_contents = requests.get(url).content
+        with open(sqlitefile, "w") as f:
```
Comment (cclauss) [inline openlibrary/admin/numbers.py:135]:
```suggestion
        sqlite_contents = requests.get(url).text
        with open(sqlitefile, "w") as f:
        # -- or --
        sqlite_contents = requests.get(url).content
        with open(sqlitefile, "wb") as f:        
```

File: openlibrary/coverstore/utils.py [hunk 2]
```diff
@@ -77,9 +78,8 @@ def ol_get(olkey):
 
 
 def download(url):
-    req = Request(url, headers={'User-Agent': USER_AGENT})
-    r = urlopen(req)
-    return r.read()
+    response = requests.get(url, headers={'User-Agent': USER_AGENT})
+    return response.content
```
Comment (cclauss) [inline openlibrary/coverstore/utils.py:82]:
```suggestion
    return requests.get(url, headers={'User-Agent': USER_AGENT}).content
```

File: openlibrary/plugins/admin/code.py [hunk 3]
```diff
@@ -387,7 +388,7 @@ def get_stats(self, today):
 class ipstats:
     def GET(self):
         web.header('Content-Type', 'application/json')
-        json = urllib.request.urlopen("http://www.archive.org/download/stats/numUniqueIPsOL.json").read()
+        json = requests.get("http://www.archive.org/download/stats/numUniqueIPsOL.json").content
         return delegate.RawText(json)
```
Comment (cclauss) [inline openlibrary/plugins/admin/code.py:392]:
```suggestion
        json_bytes = requests.get("http://www.archive.org/download/stats/numUniqueIPsOL.json").content
        return delegate.RawText(json_bytes)
```

Comment (aasifkhan7) [reply openlibrary/plugins/admin/code.py:392]: @cclauss I am actually not sure how to check what `delegate.RawTest()` is expecting. Can you please let me know how to go on about checking it?

Comment (cclauss) [reply openlibrary/plugins/admin/code.py:392]: I would say that we want `.text`

% `grep -e RawText -e rawtext **/*.py`
```
infogami/core/code.py:        return delegate.RawText(render.feed(site, changes))
infogami/plugins/api/code.py:        return delegate.RawText(out, content_type=content_type)
infogami/utils/app.py:        from infogami.utils.delegate import RawText
infogami/utils/app.py:        return RawText(simplejson.dumps(data))
infogami/utils/delegate.py:        out = RawText("")
infogami/utils/delegate.py:    if hasattr(out, 'rawtext'):
infogami/utils/delegate.py:        html = out.rawtext
infogami/utils/delegate.py:class RawText(web.storage):
infogami/utils/delegate.py:        web.storage.__init__(self, rawtext=text, **kw)
```

Comment (aasifkhan7) [reply openlibrary/plugins/admin/code.py]: Yes. 
```
infogami/utils/delegate.py:        out = RawText("")
```
This makes it very clear. Thanks @cclauss 
Made the changes.

File: openlibrary/admin/numbers.py [hunk 4]
```diff
@@ -132,10 +131,9 @@ def admin_range__visitors(**kargs):
         sqlitefile = tempfile.mktemp(prefix="sqlite-")
         url = "http://www.archive.org/download/stats/numUniqueIPsOL.sqlite"
         logging.debug("  Downloading '%s'", url)
-        sqlite_contents = urllib.request.urlopen(url).read()
-        f = open(sqlitefile, "w")
-        f.write(sqlite_contents)
-        f.close()
+        sqlite_contents = requests.get(url).content
+        with open(sqlitefile, "wb") as f:
+            f.write(sqlite_contents)
```
[the remark this answers is not in the document]
Comment (aasifkhan7) [reply openlibrary/admin/numbers.py]: Done. @cclauss

**Review by mekarpeles — APPROVED** [review]
lgtm as well

**Pull request merged**